### ISSUE TITLE
0x8badf00d watchdog crash when app start

### ISSUE THREAD

**Issue by fanwise** [firebase/firebase-ios-sdk#5708]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.3
  * Firebase SDK version: 6.23.0
  * Firebase Component: (Analytics, Crashlytics, Performance)
  * Component version: 6.23.0
  * Installation method: CocoaPods

#### Steps to reproduce:

This is a crash that happend by chance when app start. Most users are fine, but some report this crash, and this crash was not been reported to firebase.
We got one crash report from a user's device, hope that will help.

In the report I see it happened because `dispatch_sync` in `[GDTCORUploadCoordinator startTimer]` block the main thread, and the watchdog kill the app. 

And in the `com.google.GDTCORFlatFileStorage` queue (Thread 12) the `dispatch_sync` in `[GDTCORUploadCoordinator initWithCoder:]` also been locked. They both wait for the `com.google.GDTCORUploadCoordinator` queue.

The things happened in the `com.google.GDTCORUploadCoordinator` queue (Thread 15), was blocked by `dispatch_sync` in `[GDTCCTPrioritizer eventsForTarget:]`. So I think why all these happened is beause the `[GDTCCTPrioritizer eventsForTarget:]` could not run normally.

For further dig, I found that the code in `[GDTCCTPrioritizer eventsForTarget:]` should run in `com.google.GDTCCTPrioritizer` queue, but that not showed in the report. And more strange, I found many things happened in `com.google.GDTCOREventIDQueue` queue (Thread 14), which should not run here.

In GDTCOREvent source code, I see just writeToFile run in `com.google.GDTCOREventIDQueue` queue.

```
dispatch_sync(eventIDQueue, ^{
  result = @(nextEventID);
  nextEventID++;
  NSError *error;
  [[result stringValue] writeToFile:counterPath
                         atomically:YES
                           encoding:NSUTF8StringEncoding
                              error:&error];
  GDTCORAssert(error == nil, @"There was an error saving the new counter value to disk.");
});
```

I was confused here, and could not go any further.

Hope you know why this happened, and give a solution.

Below is the crash report and podfile.lock.

```
{"app_cohort":"2|date=1590346800000&sf=143465&tid=942a08e37c6f8c4fb88527317dba99bfc99e3f021ecb48900f63b11bdcbaeb2d&ttype=e","app_name":"MyApp","app_version":"7.15.0","timestamp":"2020-05-27 12:40:47.00 +0800","slice_uuid":"96e0081e-a5a4-3e82-bdb3-2df4657b0e9b","adam_id":499819758,"build_version":"3235","bundleID":"com.MyAppchina.myMyAppmoments","share_with_app_devs":1,"is_first_party":0,"bug_type":"109","os_version":"iPhone OS 13.4.1 (17E262)","incident_id":"41E11E80-F2DC-46DC-8037-F333E35AD08E","name":"MyApp","is_beta":1}
Incident Identifier: 41E11E80-F2DC-46DC-8037-F333E35AD08E
Beta Identifier:     F2D42EAC-0FD5-4741-BE1B-B919A3FFFDBB
Hardware Model:      iPhone11,2
Process:             MyApp [6521]
Path:                /private/var/containers/Bundle/Application/650D061D-56EC-494E-8455-6291AF1DE59A/MyApp.app/MyApp
Identifier:          com.MyAppchina.myMyAppmoments
Version:             3235 (7.15.0)
AppStoreTools:       11E608a
AppVariant:          1:iPhone11,2:13
Beta:                YES
Code Type:           ARM-64 (Native)
Role:                Non UI
Parent Process:      launchd [1]
Coalition:           com.MyAppchina.myMyAppmoments [1180]


Date/Time:           2020-05-27 12:40:46.0638 +0800
Launch Time:         2020-05-27 12:37:59.8726 +0800
OS Version:          iPhone OS 13.4.1 (17E262)
Release Type:        User
Baseband Version:    2.05.13
Report Version:      104

Exception Type:  EXC_CRASH (SIGKILL)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: Namespace SPRINGBOARD, Code 0x8badf00d
Termination Description: SPRINGBOARD, scene-update watchdog transgression: application<com.MyAppchina.myMyAppmoments>:6521 exhausted real (wall clock) time allowance of 10.00 seconds | ProcessVisibility: Background | ProcessState: Running | WatchdogEvent: scene-update | WatchdogVisibility: Foreground | WatchdogCPUStatistics: ( | "Elapsed total CPU time (seconds): 52.090 (user 52.090, system 0.000), 14% CPU", | "Elapsed application CPU time (seconds): 16.597, 4% CPU" | )
Triggered by Thread:  0

Thread 0 name:  Dispatch queue: com.apple.main-thread
Thread 0 Crashed:
0   libsystem_kernel.dylib        	0x00000001a3948920 kevent_id + 8
1   libdispatch.dylib             	0x00000001a381d5c0 _dispatch_kq_poll + 220
2   libdispatch.dylib             	0x00000001a381dff8 _dispatch_event_loop_wait_for_ownership + 404
3   libdispatch.dylib             	0x00000001a380d490 __DISPATCH_WAIT_FOR_QUEUE__ + 308
4   libdispatch.dylib             	0x00000001a380d098 _dispatch_sync_f_slow + 148
5   MyApp                     	    0x0000000100900fc8 -[GDTCORUploadCoordinator startTimer] + 76
6   MyApp                     	    0x00000001008fd2ec -[GDTCORLifecycle applicationWillEnterForeground:] + 340
7   CoreFoundation                	0x00000001a3ab81e0 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 28
8   CoreFoundation                	0x00000001a3ab8230 ___CFXRegistrationPost1_block_invoke + 68
9   CoreFoundation                	0x00000001a3ab7520 _CFXRegistrationPost1 + 396
10  CoreFoundation                	0x00000001a3ab71d4 ___CFXNotificationPost_block_invoke + 92
11  CoreFoundation                	0x00000001a3a30678 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1420
12  CoreFoundation                	0x00000001a3ab6b14 _CFXNotificationPost + 1260
13  Foundation                    	0x00000001a3e1df2c -[NSNotificationCenter postNotificationName:object:userInfo:] + 64
14  MyApp                     	    0x00000001008fe804 -[GDTCORApplication iOSApplicationWillEnterForeground:] + 112
15  CoreFoundation                	0x00000001a3ab81e0 __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 28
16  CoreFoundation                	0x00000001a3ab8230 ___CFXRegistrationPost1_block_invoke + 68
17  CoreFoundation                	0x00000001a3ab7520 _CFXRegistrationPost1 + 396
18  CoreFoundation                	0x00000001a3ab71d4 ___CFXNotificationPost_block_invoke + 92
19  CoreFoundation                	0x00000001a3a30678 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1420
20  CoreFoundation                	0x00000001a3ab6b14 _CFXNotificationPost + 1260
21  Foundation                    	0x00000001a3e1df2c -[NSNotificationCenter postNotificationName:object:userInfo:] + 64
22  UIKitCore                     	0x00000001a7cc0844 -[UIApplication _sendWillEnterForegroundCallbacks] + 284
23  UIKitCore                     	0x00000001a741be14 __101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke_2 + 1780
24  UIKitCore                     	0x00000001a78e5510 _UIScenePerformActionsWithLifecycleActionMask + 112
25  UIKitCore                     	0x00000001a741b684 __101-[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:]_block_invoke + 212
26  UIKitCore                     	0x00000001a741b0b0 -[_UISceneLifecycleMultiplexer _performBlock:withApplicationOfDeactivationReasons:fromReasons:] + 296
27  UIKitCore                     	0x00000001a741b4a0 -[_UISceneLifecycleMultiplexer _evalTransitionToSettings:fromSettings:forceExit:withTransitionStore:] + 756
28  UIKitCore                     	0x00000001a741ad28 -[_UISceneLifecycleMultiplexer uiScene:transitionedFromState:withTransitionContext:] + 340
29  UIKitCore                     	0x00000001a741f444 __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke_2 + 196
30  UIKitCore                     	0x00000001a7804080 +[BSAnimationSettings+ 6058112 (UIKit) tryAnimatingWithSettings:actions:completion:] + 856
31  UIKitCore                     	0x00000001a78ff1b4 _UISceneSettingsDiffActionPerformChangesWithTransitionContext + 260
32  UIKitCore                     	0x00000001a741f180 __186-[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:]_block_invoke + 156
33  UIKitCore                     	0x00000001a78ff09c _UISceneSettingsDiffActionPerformActionsWithDelayForTransitionContext + 108
34  UIKitCore                     	0x00000001a741efd8 -[_UIWindowSceneFBSSceneTransitionContextDrivenLifecycleSettingsDiffAction _performActionsForUIScene:withUpdatedFBSScene:settingsDiff:fromSettings:transitionContext:lifecycleActionType:] + 392
35  UIKitCore                     	0x00000001a7284db8 __64-[UIScene scene:didUpdateWithDiff:transitionContext:completion:]_block_invoke + 640
36  UIKitCore                     	0x00000001a72838b8 -[UIScene _emitSceneSettingsUpdateResponseForCompletion:afterSceneUpdateWork:] + 256
37  UIKitCore                     	0x00000001a7284ae8 -[UIScene scene:didUpdateWithDiff:transitionContext:completion:] + 236
38  UIKitCore                     	0x00000001a782aee8 -[UIApplicationSceneClientAgent scene:handleEvent:withCompletion:] + 480
39  FrontBoardServices            	0x00000001a8dc16e4 -[FBSSceneImpl updater:didUpdateSettings:withDiff:transitionContext:completion:] + 544
40  FrontBoardServices            	0x00000001a8de76e8 __88-[FBSWorkspaceScenesClient sceneID:updateWithSettingsDiff:transitionContext:completion:]_block_invoke_2 + 136
41  FrontBoardServices            	0x00000001a8dcbaac -[FBSWorkspace _calloutQueue_executeCalloutFromSource:withBlock:] + 240
42  FrontBoardServices            	0x00000001a8de7604 __88-[FBSWorkspaceScenesClient sceneID:updateWithSettingsDiff:transitionContext:completion:]_block_invoke + 200
43  libdispatch.dylib             	0x00000001a380033c _dispatch_client_callout + 20
44  libdispatch.dylib             	0x00000001a38030d4 _dispatch_block_invoke_direct + 264
45  FrontBoardServices            	0x00000001a8e0d2b4 __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 48
46  FrontBoardServices            	0x00000001a8e0cf60 -[FBSSerialQueue _queue_performNextIfPossible] + 432
47  FrontBoardServices            	0x00000001a8e0d4cc -[FBSSerialQueue _performNextFromRunLoopSource] + 32
48  CoreFoundation                	0x00000001a3adb860 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
49  CoreFoundation                	0x00000001a3adb7b4 __CFRunLoopDoSource0 + 84
50  CoreFoundation                	0x00000001a3adaf04 __CFRunLoopDoSources0 + 196
51  CoreFoundation                	0x00000001a3ad5ca4 __CFRunLoopRun + 796
52  CoreFoundation                	0x00000001a3ad5660 CFRunLoopRunSpecific + 480
53  GraphicsServices              	0x00000001adee6604 GSEventRunModal + 164
54  UIKitCore                     	0x00000001a7caa15c UIApplicationMain + 1944
55  MyApp                     	    0x00000001003ad024 main + 20516 (AppDelegate.swift:27)
56  libdyld.dylib                 	0x00000001a39511ec start + 4

Thread 1 name:  com.apple.uikit.eventfetch-thread
Thread 1:
0   libsystem_kernel.dylib        	0x00000001a3924784 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001a3923ba8 mach_msg + 76
2   CoreFoundation                	0x00000001a3adb080 __CFRunLoopServiceMachPort + 152
3   CoreFoundation                	0x00000001a3ad5e0c __CFRunLoopRun + 1156
4   CoreFoundation                	0x00000001a3ad5660 CFRunLoopRunSpecific + 480
5   Foundation                    	0x00000001a3e1f6a8 -[NSRunLoop+ 34472 (NSRunLoop) runMode:beforeDate:] + 232
6   Foundation                    	0x00000001a3e1f580 -[NSRunLoop+ 34176 (NSRunLoop) runUntilDate:] + 92
7   UIKitCore                     	0x00000001a7d4f420 -[UIEventFetcher threadMain] + 156
8   Foundation                    	0x00000001a3f597a0 __NSThread__start__ + 864
9   libsystem_pthread.dylib       	0x00000001a38658fc _pthread_start + 168
10  libsystem_pthread.dylib       	0x00000001a386d9d4 thread_start + 8

Thread 2 name:  GAIThread
Thread 2:
0   libsystem_kernel.dylib        	0x00000001a3924784 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001a3923ba8 mach_msg + 76
2   CoreFoundation                	0x00000001a3adb080 __CFRunLoopServiceMachPort + 152
3   CoreFoundation                	0x00000001a3ad5e0c __CFRunLoopRun + 1156
4   CoreFoundation                	0x00000001a3ad5660 CFRunLoopRunSpecific + 480
5   Foundation                    	0x00000001a3e1f6a8 -[NSRunLoop+ 34472 (NSRunLoop) runMode:beforeDate:] + 232
6   Foundation                    	0x00000001a3e5a57c -[NSRunLoop+ 275836 (NSRunLoop) run] + 92
7   MyApp                     	    0x00000001003d6e50 +[GAI threadMain:] + 64
8   Foundation                    	0x00000001a3f597a0 __NSThread__start__ + 864
9   libsystem_pthread.dylib       	0x00000001a38658fc _pthread_start + 168
10  libsystem_pthread.dylib       	0x00000001a386d9d4 thread_start + 8

Thread 3 name:  JavaScriptCore bmalloc scavenger
Thread 3:
0   libsystem_kernel.dylib        	0x00000001a3946bc0 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	0x00000001a38691e4 _pthread_cond_wait + 680
2   libc++.1.dylib                	0x00000001a399aca8 std::__1::condition_variable::wait+ 52392 (std::__1::unique_lock<std::__1::mutex>&) + 28
3   JavaScriptCore                	0x00000001b2e64e88 void std::__1::condition_variable_any::wait<std::__1::unique_lock<bmalloc::Mutex> >+ 577160 (std::__1::unique_lock<bmalloc::Mutex>&) + 104
4   JavaScriptCore                	0x00000001b2e68c34 bmalloc::Scavenger::threadRunLoop+ 592948 () + 156
5   JavaScriptCore                	0x00000001b2e68958 bmalloc::Scavenger::Scavenger+ 592216 (std::__1::lock_guard<bmalloc::Mutex> const&) + 0
6   JavaScriptCore                	0x00000001b2e69bd0 std::__1::__thread_specific_ptr<std::__1::__thread_struct>::set_pointer+ 596944 (std::__1::__thread_struct*) + 0
7   libsystem_pthread.dylib       	0x00000001a38658fc _pthread_start + 168
8   libsystem_pthread.dylib       	0x00000001a386d9d4 thread_start + 8

Thread 4 name:  WebThread
Thread 4:
0   libsystem_kernel.dylib        	0x00000001a3924784 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001a3923ba8 mach_msg + 76
2   CoreFoundation                	0x00000001a3adb080 __CFRunLoopServiceMachPort + 152
3   CoreFoundation                	0x00000001a3ad5e0c __CFRunLoopRun + 1156
4   CoreFoundation                	0x00000001a3ad5660 CFRunLoopRunSpecific + 480
5   WebCore                       	0x00000001ac2a88d4 RunWebThread+ 10594516 (void*) + 596
6   libsystem_pthread.dylib       	0x00000001a38658fc _pthread_start + 168
7   libsystem_pthread.dylib       	0x00000001a386d9d4 thread_start + 8

Thread 5 name:  com.apple.CFSocket.private
Thread 5:
0   libsystem_kernel.dylib        	0x00000001a394707c __select + 8
1   CoreFoundation                	0x00000001a3ae923c __CFSocketManager + 620
2   libsystem_pthread.dylib       	0x00000001a38658fc _pthread_start + 168
3   libsystem_pthread.dylib       	0x00000001a386d9d4 thread_start + 8

Thread 6 name:  com.google.firebase.crashlytics.MachExceptionServer
Thread 6:
0   libsystem_kernel.dylib        	0x00000001a3924784 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001a3923ba8 mach_msg + 76
2   MyApp                     	    0x0000000100736028 FIRCLSMachExceptionServer + 104
3   libsystem_pthread.dylib       	0x00000001a38658fc _pthread_start + 168
4   libsystem_pthread.dylib       	0x00000001a386d9d4 thread_start + 8

Thread 7 name:  com.apple.CoreMotion.MotionThread
Thread 7:
0   libsystem_kernel.dylib        	0x00000001a3924784 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001a3923ba8 mach_msg + 76
2   CoreFoundation                	0x00000001a3adb080 __CFRunLoopServiceMachPort + 152
3   CoreFoundation                	0x00000001a3ad5e0c __CFRunLoopRun + 1156
4   CoreFoundation                	0x00000001a3ad5660 CFRunLoopRunSpecific + 480
5   CoreFoundation                	0x00000001a3ad63b0 CFRunLoopRun + 64
6   CoreMotion                    	0x00000001b0ad7608 0x1b09e9000 + 976392
7   libsystem_pthread.dylib       	0x00000001a38658fc _pthread_start + 168
8   libsystem_pthread.dylib       	0x00000001a386d9d4 thread_start + 8

Thread 8 name:  org.hxhg.jiguang.thread
Thread 8:
0   libsystem_kernel.dylib        	0x00000001a3924784 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001a3923ba8 mach_msg + 76
2   CoreFoundation                	0x00000001a3adb080 __CFRunLoopServiceMachPort + 152
3   CoreFoundation                	0x00000001a3ad5e0c __CFRunLoopRun + 1156
4   CoreFoundation                	0x00000001a3ad5660 CFRunLoopRunSpecific + 480
5   CoreFoundation                	0x00000001a3ad63b0 CFRunLoopRun + 64
6   MyApp                     	    0x00000001004488e8 -[JPUSHThread clientThreadMain] + 100
7   Foundation                    	0x00000001a3f597a0 __NSThread__start__ + 864
8   libsystem_pthread.dylib       	0x00000001a38658fc _pthread_start + 168
9   libsystem_pthread.dylib       	0x00000001a386d9d4 thread_start + 8

Thread 9 name:  com.apple.NSURLConnectionLoader
Thread 9:
0   libsystem_kernel.dylib        	0x00000001a3924784 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001a3923ba8 mach_msg + 76
2   CoreFoundation                	0x00000001a3adb080 __CFRunLoopServiceMachPort + 152
3   CoreFoundation                	0x00000001a3ad5e0c __CFRunLoopRun + 1156
4   CoreFoundation                	0x00000001a3ad5660 CFRunLoopRunSpecific + 480
5   CFNetwork                     	0x00000001a6dfa584 0x1a6df9000 + 5508
6   Foundation                    	0x00000001a3f597a0 __NSThread__start__ + 864
7   libsystem_pthread.dylib       	0x00000001a38658fc _pthread_start + 168
8   libsystem_pthread.dylib       	0x00000001a386d9d4 thread_start + 8

Thread 10 name:  AMapNetwork
Thread 10:
0   libsystem_kernel.dylib        	0x00000001a3924784 mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001a3923ba8 mach_msg + 76
2   CoreFoundation                	0x00000001a3adb080 __CFRunLoopServiceMachPort + 152
3   CoreFoundation                	0x00000001a3ad5e0c __CFRunLoopRun + 1156
4   CoreFoundation                	0x00000001a3ad5660 CFRunLoopRunSpecific + 480
5   Foundation                    	0x00000001a3e1f6a8 -[NSRunLoop+ 34472 (NSRunLoop) runMode:beforeDate:] + 232
6   Foundation                    	0x00000001a3e5a57c -[NSRunLoop+ 275836 (NSRunLoop) run] + 92
7   AMapSDK                       	0x00000001020d9d94 0x10205c000 + 515476
8   Foundation                    	0x00000001a3f597a0 __NSThread__start__ + 864
9   libsystem_pthread.dylib       	0x00000001a38658fc _pthread_start + 168
10  libsystem_pthread.dylib       	0x00000001a386d9d4 thread_start + 8

Thread 11:
0   libsystem_pthread.dylib       	0x00000001a386d9c0 start_wqthread + 0

Thread 12 name:  Dispatch queue: com.google.GDTCORFlatFileStorage
Thread 12:
0   libsystem_kernel.dylib        	0x00000001a3948920 kevent_id + 8
1   libdispatch.dylib             	0x00000001a381d5c0 _dispatch_kq_poll + 220
2   libdispatch.dylib             	0x00000001a381dff8 _dispatch_event_loop_wait_for_ownership + 404
3   libdispatch.dylib             	0x00000001a380d490 __DISPATCH_WAIT_FOR_QUEUE__ + 308
4   libdispatch.dylib             	0x00000001a380d098 _dispatch_sync_f_slow + 148
5   MyApp                     	    0x00000001009017e8 -[GDTCORUploadCoordinator initWithCoder:] + 148
6   Foundation                    	0x00000001a3f21d8c _decodeObjectBinary + 2248
7   Foundation                    	0x00000001a3f2122c _decodeObject + 344
8   Foundation                    	0x00000001a3e29214 -[NSKeyedUnarchiver decodeObjectForKey:] + 156
9   Foundation                    	0x00000001a3e29078 -[NSKeyedUnarchiver decodeObjectOfClasses:forKey:] + 356
10  MyApp                     	    0x00000001008fcb3c -[GDTCORFlatFileStorage initWithCoder:] + 680
11  Foundation                    	0x00000001a3f21d8c _decodeObjectBinary + 2248
12  Foundation                    	0x00000001a3f2122c _decodeObject + 344
13  Foundation                    	0x00000001a3e29214 -[NSKeyedUnarchiver decodeObjectForKey:] + 156
14  Foundation                    	0x00000001a3e29078 -[NSKeyedUnarchiver decodeObjectOfClasses:forKey:] + 356
15  Foundation                    	0x00000001a3e28cd4 -[NSCoder+ 72916 (Exceptions) __tryDecodeObjectForKey:error:decodeBlock:] + 108
16  Foundation                    	0x00000001a3e28c40 -[NSCoder decodeTopLevelObjectOfClasses:forKey:error:] + 124
17  Foundation                    	0x00000001a3e27ea0 +[NSKeyedUnarchiver unarchivedObjectOfClasses:fromData:error:] + 132
18  Foundation                    	0x00000001a3e6fc00 +[NSKeyedUnarchiver unarchivedObjectOfClass:fromData:error:] + 120
19  MyApp                     	    0x00000001008fe09c GDTCORDecodeArchive + 208
20  MyApp                     	    0x00000001008fc448 __43-[GDTCORFlatFileStorage appWillForeground:]_block_invoke + 96
21  libdispatch.dylib             	0x00000001a37feec4 _dispatch_call_block_and_release + 32
22  libdispatch.dylib             	0x00000001a380033c _dispatch_client_callout + 20
23  libdispatch.dylib             	0x00000001a380685c _dispatch_lane_serial_drain + 568
24  libdispatch.dylib             	0x00000001a3807290 _dispatch_lane_invoke + 400
25  libdispatch.dylib             	0x00000001a3810928 _dispatch_workloop_worker_thread + 584
26  libsystem_pthread.dylib       	0x00000001a3867714 _pthread_wqthread + 276
27  libsystem_pthread.dylib       	0x00000001a386d9c8 start_wqthread + 8

Thread 13:
0   libsystem_pthread.dylib       	0x00000001a386d9c0 start_wqthread + 0

Thread 14 name:  Dispatch queue: com.google.GDTCOREventIDQueue
Thread 14:
0   libsystem_kernel.dylib        	0x00000001a3949ac8 write + 8
1   Foundation                    	0x00000001a3f3b4bc _NSWriteToFileDescriptorWithProgress + 220
2   Foundation                    	0x00000001a3f3cb84 ___NSWriteDataToFileWithExtendedAttributes_block_invoke + 80
3   Foundation                    	0x00000001a3e62094 -[NSData+ 307348 (NSData) enumerateByteRangesUsingBlock:] + 132
4   Foundation                    	0x00000001a3f3c6f8 _NSWriteDataToFileWithExtendedAttributes + 572
5   Foundation                    	0x00000001a3f567c0 writeStringToURLOrPath + 224
6   MyApp                     	    0x00000001008fa5a4 __26+[GDTCOREvent nextEventID]_block_invoke.9 + 164
7   libdispatch.dylib             	0x00000001a380033c _dispatch_client_callout + 20
8   libdispatch.dylib             	0x00000001a380d1f4 _dispatch_lane_barrier_sync_invoke_and_complete + 60
9   MyApp                     	    0x00000001008fa330 +[GDTCOREvent nextEventID] + 148
10  MyApp                     	    0x00000001008fa648 -[GDTCOREvent initWithMappingID:target:] + 140
11  MyApp                     	    0x00000001008fac10 -[GDTCOREvent initWithCoder:] + 288
12  Foundation                    	0x00000001a3f21d8c _decodeObjectBinary + 2248
13  Foundation                    	0x00000001a3e70280 -[NSKeyedUnarchiver _decodeArrayOfObjectsForKey:] + 868
14  Foundation                    	0x00000001a3e4710c -[NSSet+ 196876 (NSSet) initWithCoder:] + 176
15  Foundation                    	0x00000001a3f21d8c _decodeObjectBinary + 2248
16  Foundation                    	0x00000001a3f2122c _decodeObject + 344
17  Foundation                    	0x00000001a3e29214 -[NSKeyedUnarchiver decodeObjectForKey:] + 156
18  Foundation                    	0x00000001a3e29078 -[NSKeyedUnarchiver decodeObjectOfClasses:forKey:] + 356
19  MyApp                     	    0x00000001009052b8 -[GDTCCTPrioritizer initWithCoder:] + 256
20  Foundation                    	0x00000001a3f21d8c _decodeObjectBinary + 2248
21  Foundation                    	0x00000001a3f2122c _decodeObject + 344
22  Foundation                    	0x00000001a3e29214 -[NSKeyedUnarchiver decodeObjectForKey:] + 156
23  Foundation                    	0x00000001a3e29078 -[NSKeyedUnarchiver decodeObjectOfClasses:forKey:] + 356
24  Foundation                    	0x00000001a3e28cd4 -[NSCoder+ 72916 (Exceptions) __tryDecodeObjectForKey:error:decodeBlock:] + 108
25  Foundation                    	0x00000001a3e28c40 -[NSCoder decodeTopLevelObjectOfClasses:forKey:error:] + 124
26  Foundation                    	0x00000001a3e27ea0 +[NSKeyedUnarchiver unarchivedObjectOfClasses:fromData:error:] + 132
27  Foundation                    	0x00000001a3e6fc00 +[NSKeyedUnarchiver unarchivedObjectOfClass:fromData:error:] + 120
28  MyApp                     	    0x00000001008fe09c GDTCORDecodeArchive + 208
29  MyApp                     	    0x00000001009054a8 __39-[GDTCCTPrioritizer appWillForeground:]_block_invoke + 84
30  libdispatch.dylib             	0x00000001a37feec4 _dispatch_call_block_and_release + 32
31  libdispatch.dylib             	0x00000001a380033c _dispatch_client_callout + 20
32  libdispatch.dylib             	0x00000001a380685c _dispatch_lane_serial_drain + 568
33  libdispatch.dylib             	0x00000001a3807290 _dispatch_lane_invoke + 400
34  libdispatch.dylib             	0x00000001a3810928 _dispatch_workloop_worker_thread + 584
35  libsystem_pthread.dylib       	0x00000001a3867714 _pthread_wqthread + 276
36  libsystem_pthread.dylib       	0x00000001a386d9c8 start_wqthread + 8

Thread 15 name:  Dispatch queue: com.google.GDTCORUploadCoordinator
Thread 15:
0   libsystem_kernel.dylib        	0x00000001a3948920 kevent_id + 8
1   libdispatch.dylib             	0x00000001a381d5c0 _dispatch_kq_poll + 220
2   libdispatch.dylib             	0x00000001a381dff8 _dispatch_event_loop_wait_for_ownership + 404
3   libdispatch.dylib             	0x00000001a380d490 __DISPATCH_WAIT_FOR_QUEUE__ + 308
4   libdispatch.dylib             	0x00000001a380d098 _dispatch_sync_f_slow + 148
5   MyApp                     	    0x0000000100904604 -[GDTCCTPrioritizer eventsForTarget:] + 136
6   MyApp                     	    0x0000000100906bf4 -[GDTCCTUploader readyToUploadTarget:conditions:] + 108
7   MyApp                     	    0x0000000100901470 __52-[GDTCORUploadCoordinator uploadTargets:conditions:]_block_invoke + 492
8   libdispatch.dylib             	0x00000001a37feec4 _dispatch_call_block_and_release + 32
9   libdispatch.dylib             	0x00000001a380033c _dispatch_client_callout + 20
10  libdispatch.dylib             	0x00000001a380685c _dispatch_lane_serial_drain + 568
11  libdispatch.dylib             	0x00000001a3807290 _dispatch_lane_invoke + 400
12  libdispatch.dylib             	0x00000001a3810928 _dispatch_workloop_worker_thread + 584
13  libsystem_pthread.dylib       	0x00000001a3867714 _pthread_wqthread + 276
14  libsystem_pthread.dylib       	0x00000001a386d9c8 start_wqthread + 8

Thread 16:
0   libsystem_pthread.dylib       	0x00000001a386d9c0 start_wqthread + 0

Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000000000001   x1: 0x0000000000000000   x2: 0x0000000000000001   x3: 0x000000016fa544e0
    x4: 0x0000000000000001   x5: 0x0000000000000000   x6: 0x0000000000000000   x7: 0x0000000000000403
    x8: 0x0000000000000400   x9: 0x00000001f1520240  x10: 0x0000000000000400  x11: 0x00000000000020ff
   x12: 0x0000003800000000  x13: 0x000000016fa544e0  x14: 0x0000000000000084  x15: 0x0000004000000001
   x16: 0x0000000000000177  x17: 0x0000000000000001  x18: 0x0000000000000000  x19: 0x0000000000000403
   x20: 0x0000000000000000  x21: 0x0000000000000000  x22: 0x0000000000000001  x23: 0x000000016fa544e0
   x24: 0x0000000000000001  x25: 0x000000016fa544e0  x26: 0x00000002829a6c00  x27: 0x00000001f1520240
   x28: 0x0000000000020400   fp: 0x000000016fa544c0   lr: 0x00000001a381d5c0
    sp: 0x000000016fa54460   pc: 0x00000001a3948920 cpsr: 0x00000000
   esr: 0x56000080  Address size fault
```

```
  - Firebase/Analytics (6.23.0):
    - Firebase/Core
  - Firebase/Core (6.23.0):
    - Firebase/CoreOnly
    - FirebaseAnalytics (= 6.4.2)
  - Firebase/CoreOnly (6.23.0):
    - FirebaseCore (= 6.6.7)
  - Firebase/Crashlytics (6.23.0):
    - Firebase/CoreOnly
    - FirebaseCrashlytics (~> 4.0.0)
  - Firebase/Performance (6.23.0):
    - Firebase/CoreOnly
    - FirebasePerformance (~> 3.1.11)
  - FirebaseABTesting (3.2.0):
    - FirebaseAnalyticsInterop (~> 1.3)
    - FirebaseCore (~> 6.1)
    - Protobuf (>= 3.9.2, ~> 3.9)
  - FirebaseAnalytics (6.4.2):
    - FirebaseCore (~> 6.6)
    - FirebaseInstallations (~> 1.2)
    - GoogleAppMeasurement (= 6.4.2)
    - GoogleUtilities/AppDelegateSwizzler (~> 6.0)
    - GoogleUtilities/MethodSwizzler (~> 6.0)
    - GoogleUtilities/Network (~> 6.0)
    - "GoogleUtilities/NSData+zlib (~> 6.0)"
    - nanopb (= 0.3.9011)
  - FirebaseAnalyticsInterop (1.5.0)
  - FirebaseCore (6.6.7):
    - FirebaseCoreDiagnostics (~> 1.2)
    - FirebaseCoreDiagnosticsInterop (~> 1.2)
    - GoogleUtilities/Environment (~> 6.5)
    - GoogleUtilities/Logger (~> 6.5)
  - FirebaseCoreDiagnostics (1.2.4):
    - FirebaseCoreDiagnosticsInterop (~> 1.2)
    - GoogleDataTransportCCTSupport (~> 3.0)
    - GoogleUtilities/Environment (~> 6.5)
    - GoogleUtilities/Logger (~> 6.5)
    - nanopb (~> 0.3.901)
  - FirebaseCoreDiagnosticsInterop (1.2.0)
  - FirebaseCrashlytics (4.0.0):
    - FirebaseAnalyticsInterop (~> 1.2)
    - FirebaseCore (~> 6.6)
    - FirebaseInstallations (~> 1.1)
    - GoogleDataTransport (~> 6.0)
    - GoogleDataTransportCCTSupport (~> 3.0)
    - nanopb (~> 0.3.901)
    - PromisesObjC (~> 1.2)
  - FirebaseInstallations (1.2.0):
    - FirebaseCore (~> 6.6)
    - GoogleUtilities/Environment (~> 6.6)
    - GoogleUtilities/UserDefaults (~> 6.6)
    - PromisesObjC (~> 1.2)
  - FirebasePerformance (3.1.11):
    - FirebaseCore (~> 6.6)
    - FirebaseInstallations (~> 1.1)
    - FirebaseRemoteConfig (~> 4.4)
    - GoogleToolboxForMac/Logger (~> 2.1)
    - "GoogleToolboxForMac/NSData+zlib (~> 2.1)"
    - GoogleUtilities/Environment (~> 6.2)
    - GoogleUtilities/ISASwizzler (~> 6.2)
    - GoogleUtilities/MethodSwizzler (~> 6.2)
    - GTMSessionFetcher/Core (~> 1.1)
    - Protobuf (~> 3.9)
  - FirebaseRemoteConfig (4.4.10):
    - FirebaseABTesting (~> 3.1)
    - FirebaseAnalyticsInterop (~> 1.4)
    - FirebaseCore (~> 6.2)
    - FirebaseInstallations (~> 1.1)
    - GoogleUtilities/Environment (~> 6.2)
    - "GoogleUtilities/NSData+zlib (~> 6.2)"
    - Protobuf (>= 3.9.2, ~> 3.9)
```

#### Relevant Code:

It's not a always happened crash, so I don't has a reproduce code.

**Comment by maksymmalyhin** [comment]
This one looks similar to #5707.

**Comment by memoryxy** [comment]
Have same problem too. need be fixed.

**Comment by jiehu5114** [comment]
same issue

**Comment by mikehaney24** [comment]
This should be fixed in the updated version of GoogleDataTransport, 6.2.1. If you're still having an issue, please open a new issue with your Podfile.lock and the error messages you're seeing.

**Comment by gilrodar** [comment]
> This should be fixed in the updated version of GoogleDataTransport, 6.2.1. If you're still having an issue, please open a new issue with your Podfile.lock and the error messages you're seeing.

Yes, my problem is solved with version 6.2.1. Thanks!!

**Comment by seenitatulgupta** [comment]
Same issue happened for me on iPhone 11 Pro Max running on iOS version 13.5.1

Here are my cocoa pods version for Firebase:
Using Firebase (6.26.0)
Using FirebaseABTesting (3.2.0)
Using FirebaseAnalytics (6.6.0)
Using FirebaseAnalyticsInterop (1.5.0)
Using FirebaseCore (6.7.2)
Using FirebaseCoreDiagnostics (1.3.0)
Using FirebaseCoreDiagnosticsInterop (1.2.0)
Using FirebaseCrashlytics (4.1.1)
Using FirebaseInstallations (1.3.0)
Using FirebaseInstanceID (4.3.4)
Using FirebaseMessaging (4.4.1)
Using FirebasePerformance (3.1.11)
Using FirebaseRemoteConfig (4.5.0)
Using GTMSessionFetcher (1.4.0)
Using GoogleAppMeasurement (6.6.0)
Using GoogleDataTransport (6.2.1)
Using GoogleDataTransportCCTSupport (3.1.0)
Using GoogleToolboxForMac (2.2.2)
Using GoogleUtilities (6.6.0)
